Blish HUD/GameServices/ArcDpsService.cs

```diff
@@ -1,96 +1,122 @@
 ﻿using System;
 using System.Diagnostics;
-using System.IO.Pipes;
+using System.Net;
 using System.Threading;
-using System.Threading.Tasks;
+using Blish_HUD.ArcDps;
+using Blish_HUD.ArcDps.Models;
 using Microsoft.Xna.Framework;
-using ProtoBuf;
 
 namespace Blish_HUD
 {
     public class ArcDpsService : GameService
     {
-        public bool ArcPresent;
-        private CancellationToken Cancel;
-        public bool HudIsActive;
-        private Mutex Mut;
-        private NamedPipeServerStream Server;
-        private Stopwatch Watch;
+        public delegate void RawCombatEvent(CombatEvent data);
+
+        private static readonly object WatchLock = new object();
+        private readonly TimeSpan _leeway = TimeSpan.FromMilliseconds(1000);
+        private bool _hudIsActive;
+#if DEBUG
+        public static long Counter = 0;
+#endif
+
+        private SocketListener _server;
+
+        private Stopwatch _stopwatch;
+
+        /// <summary>
+        ///     Indicates if arcdps updated <see cref="HudIsActive" /> in the last second (it should every in-game frame)
+        /// </summary>
+        public bool RenderPresent { get; private set; }
+
+        /// <summary>
+        ///     Indicates if arcdps currently draws its HUD (not in character select, cut scenes or loading screens)
+        /// </summary>
+        public bool HudIsActive
+        {
+            get
+            {
+                lock (WatchLock)
+                {
+                    return _hudIsActive;
+                }
+            }
+            private set
+            {
+                lock (WatchLock)
+                {
+                    _stopwatch.Restart();
+                    _hudIsActive = value;
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Holds unprocessed combat data
+        /// </summary>
+        public event RawCombatEvent OnRawCombatEvent;
 
         protected override void Initialize()
         {
-            Server = new NamedPipeServerStream("BHUDrender");
-            Mut = new Mutex(false);
+            _stopwatch = new Stopwatch();
+            _server = new SocketListener(10, 200_000);
+            _server.ReceivedMessage += MessageHandler;
+#if DEBUG
+            OnRawCombatEvent += data =>
+            {
+                Interlocked.Increment(ref Counter);
```

Comment:
`Interlocked.Read()` isn't necessary on 64-bit systems, right?  If this isn't used other than in the counters rendered I am not sure it is necessary since the UI is already refreshing so quickly that getting the absolute latest value of the counter may not be necessary.

Reply:
the whole counter is just for debugging while this is broadly untested. it will defintely get removed in the future. i currently dont know how i would want to approach that in the current state. im open for suggestions

Reply:
After reading more about it, I'm thinking my complaints are not that big of a deal.  If we find there is an issue later, we can change it before merging into [master](../../tree/master).